pkgs/os-specific/linux/asusctl/default.nix

```diff
@@ -0,0 +1,42 @@
+{ stdenv, lib, fetchFromGitLab, rustPlatform, rust
+, pkg-config
+, udev
+}:
+
+with rustPlatform;
+
+buildRustPackage rec {
+  pname = "asusctl";
+  version = "4.3.0";
+
+  src = fetchFromGitLab {
+    owner = "asus-linux";
+    repo = "asusctl";
+    rev = version;
+    hash = "sha256-s0y5oK079Mwg9RTbSbcW1jGjhRH0X/GnSPI1e/G8m1c=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ udev ];
+
+  cargoHash = "sha256-lKviCcwuJaLx+H9Qu6XXUFM3L99A4kgP1vR46hBhVRE=";
+
+  postPatch = ''
+    # The build is placed under target/x86_64-unknown-linux-gnu/release, while Makefile expects it under
+    # target/release. We'll create target/release to be compatible with Makefile.
+    substituteInPlace Makefile \
+      --replace "./target/release/" "./target/${rust.toRustTargetSpec stdenv.hostPlatform}/release/"
+  '';
+
+  installPhase = ''
+    make install DESTDIR="$out" prefix=""
+  '';
```

Comment:
buildRustPackage can copy all the build artifacts to `$out/bin`, so you can set `INSTALL_PROGRAM=true` without having to patch Makefile, and without depending on `rust` and `stdenv`.

Not really needed, but `INSTALL_DATA="install -D -m 0444"` would be nicer since you can't directly write to a nix path anyway.

Reply:
I'm not 100% sure which approach I liked better, but it does make sense to let `buildRustPackage` do its thing. I added a comment explaining that `make install` is only executed for the data :+1: I used `preInstall` for that so that `buildRustPackage`'s hooks is used in `installPhase`.

Nice find on `rust` and `stdenv`. I removed them, as they were indeed unused.

Thanks for the suggestions!